packages/material-ui/src/Menu/Menu.js

```diff
@@ -47,20 +55,45 @@ const Menu = React.forwardRef(function Menu(props, ref) {
     open,
     PaperProps = {},
     PopoverClasses,
+    setParentOpenSubMenuIndex,
     transitionDuration = 'auto',
     variant = 'selectedMenu',
-    TransitionProps: { onEntering, ...TransitionProps } = {},
+    TransitionProps: { onEnter, onEntering, onEntered, ...TransitionProps } = {},
     ...other
   } = props;
   const theme = useTheme();
 
+  const [openSubMenuIndex, setOpenSubMenuIndex] = React.useState(null);
+  const [entering, setEntering] = React.useState(false);
+  const isSubMenu = typeof setParentOpenSubMenuIndex !== 'undefined';
+
+  const atLeastOneSubMenu = React.useMemo(() => {
```

Comment:
What's the reason for memoizing this? Perf is only an argument if you measured its impact.

Reply:
It seems prudent to memoize potentially costly array traversals, especially when the cost of doing so is minimal. We can't know how many children a user will pass to a Menu. It may be premature optimization, as I haven't tested the performance impact, I suppose, but it feels like the right thing to do considering the operation being memoized. Would you rather wait until someone hits a slowdown and it can be traced to this traversal?

Reply:
> Would you rather wait until someone hits a slowdown and it can be traced to this traversal?

Yes. From my experience, almost everytime someone told me they "feel" like code is slow, it never is. Or its slowness is negligible compared to the rest of the code.

Reply:
Alright, I got rid of the memo and pushed.